src/stancjs/stancjs.ml

```diff
@@ -22,29 +42,42 @@ let stan2cpp model_name model_string =
           "Semantic check failed but reported no errors. This should never \
            happen."
   in
-  Result.bind ast
-    ~f:
-      (Fn.compose semantic_err_to_string Semantic_check.semantic_check_program)
-  |> Result.map ~f:(fun typed_ast ->
-         let mir = Ast_to_Mir.trans_prog model_name typed_ast in
-         let tx_mir = Transform_Mir.trans_prog mir in
-         let cpp = Fmt.strf "%a" Stan_math_code_gen.pp_prog tx_mir in
-         cpp )
+  if is_flag_set "--version" then Result.Ok (Fmt.strf "%s" version, [])
+  else
+    Result.bind ast
+      ~f:
+        (Fn.compose semantic_err_to_string
+           Semantic_check.semantic_check_program)
+    |> Result.map ~f:(fun typed_ast ->
+           let mir = Ast_to_Mir.trans_prog model_name typed_ast in
+           let tx_mir = Transform_Mir.trans_prog mir in
+           let opt_mir =
+             if is_flag_set "--O" then Optimize.optimization_suite tx_mir
+             else tx_mir
+           in
+           let cpp = Fmt.strf "%a" Stan_math_code_gen.pp_prog opt_mir in
+           if is_flag_set "--warn-uninitialized" then
+             Pedantic_analysis.print_warn_uninitialized mir ;
+           if is_flag_set "--warn-pedantic" then
+             Pedantic_analysis.print_warn_pedantic mir ;
```

Comment:
For now these warnings are all printed and not returned. I want to also refactor that, but that is a separate issue/PR. We could return this as part of warnings, but we still have some other warnings that would get printed (like integer division). So a separate PR to unify that would be better.